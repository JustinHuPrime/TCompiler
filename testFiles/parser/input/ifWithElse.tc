module foo;

void bar(bool b) {
  int i;
  if (b)
    i = 1;
  else
    i = 0;
}