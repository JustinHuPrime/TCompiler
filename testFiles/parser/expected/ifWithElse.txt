testFiles/parser/ifWithElse.tc (code):
FILE(1, 1, STAB(ENTRY(bar, FUNCTION(testFiles/parser/ifWithElse.tc, 3, 1, void(bool)))), MODULE(1, 1, ID(1, 8, foo, REFERENCES())), FUNDEFN(3, 1, KEYWORDTYPE(3, 1, void), ID(3, 6, bar, REFERENCES(testFiles/parser/ifWithElse.tc, 3, 1)), KEYWORDTYPE(3, 10, bool), ID(3, 15, b, REFERENCES()), STAB(ENTRY(b, VARIABLE(testFiles/parser/ifWithElse.tc, 3, 10, bool))), COMPOUNDSTMT(3, 18, STAB(ENTRY(i, VARIABLE(testFiles/parser/ifWithElse.tc, 4, 7, int))), VARDEFNSTMT(4, 3, KEYWORDTYPE(4, 3, int), ID(4, 7, i, REFERENCES(testFiles/parser/ifWithElse.tc, 4, 7))), IFSTMT(5, 3, ID(5, 7, b, REFERENCES(testFiles/parser/ifWithElse.tc, 3, 10)), EXPRESSIONSTMT(6, 5, BINOPEXP(6, 5, ASSIGN, ID(6, 5, i, REFERENCES(testFiles/parser/ifWithElse.tc, 4, 7)), LITERAL(6, 9, UBYTE(1)))), STAB(), EXPRESSIONSTMT(8, 5, BINOPEXP(8, 5, ASSIGN, ID(8, 5, i, REFERENCES(testFiles/parser/ifWithElse.tc, 4, 7)), LITERAL(8, 9, UBYTE(0)))), STAB()))))
